pyiron_atomistics/atomistics/structure/atoms.py

```diff
@@ -290,12 +288,16 @@ def set_species(self, value):
             value (list): A list atomistics.structure.periodic_table.ChemicalElement instances
 
         """
-        if value is None:
-            return
-        value = list(value)
-        self._species_to_index_dict = {el: i for i, el in enumerate(value)}
-        self._species = value[:]
-        self._store_elements = {el.Abbreviation: el for el in value}
+        if value is not None:
+            self._species = list(value)[:]
+
+    @property
+    def _store_elements(self) -> dict:
+        return {el.Abbreviation: el for el in self.species}
+
+    @property
+    def _species_to_index_dict(self) -> dict:
+        return {el: i for i, el in enumerate(self.species)}
```

Comment:
This will change the species numbering if you add or delete atoms of different types, I think.

Reply:
But whenever atoms are added or deleted, `species` is always set anew, so it shouldn’t matter

Reply:
I still couldn't test the code, but it sounds to me like it could trigger behavior like [this](https://github.com/pyiron/pyiron_atomistics/issues/718).

Reply:
I guess it's more error prone if this change is **not** merged (because then there are dangling unchecked variables). But feel free to take your time.